plugin.cpp

```diff
@@ -126,24 +129,23 @@ PLUGIN_HANDLE plugin_init(ConfigCategory* config,
 	info->configCatName = config->getName();
 	Python27Filter *pyFilter = info->handle;
 
-	// Check first we have a Python script to load
-	if (!pyFilter->setScriptName())
+	// Embedded Python 2.7 initialisation
+	if (!Py_IsInitialized())
 	{
-		// Force disable
-		pyFilter->disableFilter();
-
-		// Return filter handle
-		return (PLUGIN_HANDLE)pyFilter;
+		Py_Initialize();
+		PyEval_InitThreads(); // Initialize and acquire the global interpreter lock (GIL)
+		PyThreadState* save = PyEval_SaveThread(); // release GIL
+		pythonInitialised = true;
 	}
 
-	// Embedded Python 2.7 program name
-        Py_SetProgramName((char *)config->getName().c_str());
-	// Embedded Python 2.7 initialisation
-        Py_Initialize();
-
 	// Pass FogLAMP Data dir
 	pyFilter->setFiltersPath(getDataDir());
 
+	// Embedded Python 2.7 program name
+	Py_SetProgramName((char *)config->getName().c_str());
```

Comment:
https://docs.python.org/2/c-api/init.html - Py_SetProgramName - This function should be called before Py_Initialize() is called for the first time, if it is called at all.